### PR TITLE
Chore: Refactor pageControls and scroll handling for multi page images

### PR DESCRIPTION
**Pull request by jeremypress**

### CODE REVIEW

**Comment by jeremypress** [comment]
Need to add tests/conflicts

File: src/lib/PageControls.js [hunk 1]
```diff
@@ -79,13 +99,13 @@ class PageControls extends EventEmitter {
         // show the input box with the current page number selected within it
         this.controlsEl.classList.add(SHOW_PAGE_NUM_INPUT_CLASS);
 
-        this.pageNumInputEl.value = this.currentPageEl.textContent;
+        this.pageNumInputEl.value = this.getCurrentPageNumber();
         this.pageNumInputEl.focus();
         this.pageNumInputEl.select();
 
         // finish input when input is blurred or enter key is pressed
-        this.pageNumInputEl.addEventListener('blur', this.pageNumInputBlurHandler.bind(this));
```
Comment (jeremypress) [inline src/lib/PageControls.js]: @pramodsum bind creates a new function, which caused a new listener to get added each time this was called. It also wasn't being removed correctly as a result.

File: src/lib/PageControls.js [hunk 2]
```diff
@@ -220,7 +280,7 @@ class PageControls extends EventEmitter {
 
             case 'Escape':
                 this.hidePageNumInput();
```
Comment (jeremypress) [inline src/lib/PageControls.js]: I'll verify this on Monday but I believe there's an IE 11 issue when hitting escape in fullscreen that also closes the preview. Pretty sure this is coming from the webapp side as they own that shortcut.

File: src/lib/PageControls.js [hunk 3]
```diff
@@ -204,12 +263,13 @@ class PageControls extends EventEmitter {
         switch (key) {
             case 'Enter':
             case 'Tab':
+                this.contentEl.focus();
                 // The keycode of the 'next' key on Android Chrome is 9, which maps to 'Tab'.
-                // this.docEl.focus();
```
Comment (pramodsum) [inline src/lib/PageControls.js]: :( sorry!

File: src/lib/viewers/image/MultiImageViewer.js [hunk 4]
```diff
@@ -194,21 +199,19 @@ class MultiImageViewer extends ImageBaseViewer {
...
      */
-    bindControlListeners() {
-        super.bindControlListeners();
-
-        this.pageControls = new PageControls(this.controls, this.previousPage, this.nextPage);
-        this.pageControls.init(this.pagesCount);
-        this.pageControls.addListener('setpage', this.setPage);
+    bindPageControlListeners() {
+        this.pageControls.add(this.currentPageNumber, this.pagesCount);
+        this.pageControls.addListener('pagechange', this.setPage);
```
Comment (pramodsum) [inline src/lib/viewers/image/MultiImageViewer.js]: 👍

File: src/lib/viewers/doc/DocBaseViewer.js [hunk 5]
```diff
@@ -846,17 +850,17 @@ class DocBaseViewer extends BaseViewer {
...
-        const pageNum = event.pageNumber;
-        this.pageControls.updateCurrentPage(pageNum);
+        const pageNumber = event.pageNumber;
+        this.pageControls.updateCurrentPage(pageNumber);
 
         // We only set cache the current page if 'pagechange' was fired after
         // preview is loaded - this filters out pagechange events fired by
         // the viewer's initialization
         if (this.loaded) {
-            this.cachePage(pageNum);
```
Comment (pramodsum) [inline src/lib/viewers/doc/DocBaseViewer.js]: Are we going to add this similar functionality to the multiimage viewer?

Comment (jeremypress) [reply src/lib/viewers/doc/DocBaseViewer.js]: that's the plan!

**Pull request merged**